test/dotnet-test.Tests/AssemblyInfo.cs

```diff
@@ -0,0 +1,4 @@
+using Xunit;
+
+// Don't let the tests execute concurrently
+[assembly: CollectionBehavior(CollectionBehavior.CollectionPerAssembly)]
```

Comment:
cc @Sridhar-MS 

Well, we are only adding multiple tfm runs now, right? We should probably look into our test infrastructure (of the CLI itself), to maybe add the TFM to the path to avoid this.

I am pretty sure that that's what is causing this and this will happen for any other tests that involve multiple tfms. Even for our users. We need to figure out a way to avoid that.